File: longest_substring_without_repeating_characters/lengthOfLongestSubstring.js
const lengthOfLongestSubstring = (s) => {
	answer = 0
	current = []
	i = 0
	while (i < s.length) {
		if (!current.includes(s[i])) {
			current.push(s[i])
			i += 1
		} else {
			if (current.length > answer) { answer = current.length }
			current.shift()
		}
	}
	if (current.length > answer) { answer = current.length }
	return answer   
};

File: longest_substring_without_repeating_characters/length_of_longest_substring.rb
def length_of_longest_substring(s)
    answer = 0
    current = []
    i = 0
    while i < s.length
        if !current.include?(s[i])
            current << s[i]
            i += 1
        else
            answer = current.length if current.length > answer
            current.shift
        end
    end
    answer = current.length if current.length > answer
    answer
end
